keras_nlp/models/falcon/falcon_transformer_decoder.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from keras_nlp.backend import keras
+
+
+class FalconTransformerDecoder(keras.layers.Layer):
+    # TODO: set default values based on Falcon's config.
+    def __init__(
+        self,
+        num_attention_heads,
+        layer_norm_epsilon=1e-5,
+        attention_dropout=0,
+        feedforward_dropout=0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.num_attention_heads = num_attention_heads
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.attention_dropout = attention_dropout
+        self.feedforward_dropout = feedforward_dropout
+
+    def build(self, decoder_sequence_shape):
+        self.hidden_dim = decoder_sequence_shape[-1]
+        self._input_layernorm = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon
+        )
+        self._input_layernorm.build(decoder_sequence_shape)
+
+        # Attention layers.
+        # MHA for 1b model.
+        self.key_dim = self.hidden_dim // self.num_attention_heads
+        self._attention_layer = keras.layers.MultiHeadAttention(
+            num_heads=self.num_attention_heads,
+            key_dim=self.key_dim,
+        )
+        self._attention_layer.build(
+            decoder_sequence_shape,
+        )
+
+        self._attention_dropout = keras.layers.Dropout(
+            rate=self.attention_dropout,
+            name="attention_dropout",
+        )
+
+        self._post_attention_layernorm = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon
+        )
+        self._post_attention_layernorm.build(decoder_sequence_shape)
+
+        # Feedforward layers.
+        self._dense_h_to_4h = keras.layers.Dense(
+            units=4 * self.hidden_dim,
```

Comment:
We generally just allow a second init parameter here for the inner FF dimension. Is there a reason we can't do that here? More flexible and consistent with other keras-nlp models if we can.

Reply:
I guess I added it to make it clear to myself but I removed it to be consistent with other models.